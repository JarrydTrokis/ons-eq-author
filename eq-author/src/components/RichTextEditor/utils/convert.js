import { convertToHTML, convertFromHTML } from "draft-convert";
import { EditorState } from "draft-js";

export const toHTML = entityMap => {
  const entityToHTML = (entity, originalText) => {
    const mapper = entityMap[entity.type];
    return mapper ? mapper(entity) : originalText;
  };

  const convert = convertToHTML({ entityToHTML });

  return editorState => convert(editorState.getCurrentContent());
};

export const fromHTML = nodeToFn => {
  const htmlToEntity = (nodeName, ...otherArgs) => {
    const entity = Object.entries(nodeToFn)
      .filter(([name]) => name === nodeName)
      .map(([, fn]) => fn(nodeName, ...otherArgs))
      .find(result => result);

    return entity || null;
  };

  const convert = convertFromHTML({ htmlToEntity });

  return (html, decorator) =>
    EditorState.createWithContent(convert(html), decorator);
};
