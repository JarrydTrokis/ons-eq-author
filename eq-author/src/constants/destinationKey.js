export const NextPage = "NextPage";
export const Default = "Default";
export const EndOfCurrentSection = "EndOfCurrentSection";

export const destinationKey = {
  [EndOfCurrentSection]: "End of current section",
  [NextPage]: "Next page",
  [Default]: "Select a destination",
};
