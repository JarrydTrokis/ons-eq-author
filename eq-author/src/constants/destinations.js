export const NextPage = "NextPage";
export const Default = "Default";
export const EndOfCurrentSection = "EndOfCurrentSection";

export const logicalDestinations = () => [
  {
    id: NextPage,
    logicalDestination: NextPage,
    displayName: NextPage,
  },
  {
    id: EndOfCurrentSection,
    logicalDestination: EndOfCurrentSection,
    displayName: EndOfCurrentSection,
  },
];
