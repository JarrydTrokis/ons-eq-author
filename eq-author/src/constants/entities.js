export const SECTION = "section";
export const PAGE = "page";
export const QUESTION_CONFIRMATION = "question-confirmation";
export const INTRODUCTION = "introduction";
