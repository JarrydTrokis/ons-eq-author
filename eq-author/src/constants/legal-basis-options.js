const LEGAL_BASIS_OPTIONS = [
  {
    value: "NOTICE_1",
    description: "Section 1 of the Statistics of Trade Act 1947.",
  },
  {
    value: "NOTICE_2",
    description: "Sections 2 and 3 of the Statistics of Trade Act 1947.",
  },
  {
    value: "NOTICE_3",
    description: "Sections 3 and 4 of the Statistics of Trade Act 1947.",
  },
  {
    value: "NOTICE_NI",
    description:
      "Article 5 of the Statistics of Trade and Employment (Northern Ireland) Order 1988.",
  },
  {
    value: "NOTICE_FUELS",
    description:
      "Petroleum Act 1998 and Section 1 of the Statistics of Trade Act 1947.",
  },
  {
    value: "VOLUNTARY",
    description: "Voluntary",
  },
];

export default LEGAL_BASIS_OPTIONS;
