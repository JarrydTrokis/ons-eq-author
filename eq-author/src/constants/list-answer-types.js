export const DRIVING = "driving";
export const ANOTHER = "another";
