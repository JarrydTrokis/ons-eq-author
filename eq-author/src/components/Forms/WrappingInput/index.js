import React from "react";
import PropTypes from "prop-types";
import styled from "styled-components";
import { radius } from "constants/theme";
import AutoResizeTextArea from "react-textarea-autosize";
import withChangeHandler from "components/Forms/withChangeHandler";
import { invoke } from "lodash";
import { sharedStyles } from "components/Forms/css";
import ValidationError from "components/ValidationError";

import PasteModal, { FormatText } from "components/modals/PasteModal";

const ENTER_KEY = 13;

const StyleContext = styled.div`
  font-weight: ${(props) => (props.bold ? "bold" : "regular")};
`;

const TextArea = styled(AutoResizeTextArea)`
  ${sharedStyles};
  font-weight: inherit;
  border-radius: ${radius};
  resize: none;
  overflow: hidden; /* prevent scrollbars on Windows */
`;

class WrappingInput extends React.Component {
  static propTypes = {
    value: PropTypes.string.isRequired,
    onChange: PropTypes.func.isRequired,
    onBlur: PropTypes.func.isRequired,
    onPaste: PropTypes.func.isRequired,
    onKeyDown: PropTypes.func,
    id: PropTypes.string.isRequired,
    bold: PropTypes.bool,
    placeholder: PropTypes.string,
    errorValidationMsg: PropTypes.string,
  };

  static defaultProps = {
    bold: false,
  };

  state = { show: false, text: "", event: {} };

  handleChange = (e) => {
    e.target.value = e.target.value.replace(/\n/g, " ");
    e.target.value = e.target.value.replace(/\s+/g, " ");
    this.props.onChange(e);
  };

  handlePaste = (e) => {
    const text = e.clipboardData.getData("text");
    e.persist();
    if (/\s{2,}/g.test(text)) {
      this.setState({
        show: true,
        text: text,
        event: e,
      });
    }
  };

  handleOnPasteConfirm = () => {
    const { text, event } = this.state;
    if (event && event.persist) {
      const updatedEvent = {
        ...event,
        target: { ...event.target, value: FormatText(text) },
      };
      this.props.onChange(updatedEvent);
    }
    this.setState({ show: false, text: "" });
  };

  handleOnPasteCancel = () => {
    this.setState({ show: false, text: "" });
  };

  handleKeyDown = (e) => {
    if (e.keyCode === ENTER_KEY) {
      e.preventDefault();
    }

    invoke(this.props, "onKeyDown", e);
  };

  render() {
    const { bold, placeholder, errorValidationMsg, ...otherProps } = this.props;
    const { state } = this;
    return (
      <>
        <PasteModal
          isOpen={state.show}
          onConfirm={this.handleOnPasteConfirm}
          onCancel={this.handleOnPasteCancel}
        />
        <StyleContext
          bold={bold}
          onChange={this.handleChange}
          onKeyDown={this.handleKeyDown}
          onPaste={this.handlePaste}
        >
          <TextArea
            {...otherProps}
            placeholder={placeholder}
            invalid={errorValidationMsg}
            aria-invalid={Boolean(errorValidationMsg).toString()}
          />
          {errorValidationMsg && (
            <ValidationError>{errorValidationMsg}</ValidationError>
          )}
        </StyleContext>
      </>
    );
  }
}

export default withChangeHandler(WrappingInput);
