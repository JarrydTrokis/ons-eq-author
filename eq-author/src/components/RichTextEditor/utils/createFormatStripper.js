import { mapKeys } from "lodash";
import { filterEditorState } from "draftjs-filters";
import { filterConfig as pipedFilterConfig } from "components/RichTextEditor/entities/PipedValue";
import { filterConfig as linkFilterConfig } from "components/RichTextEditor/LinkPlugin";

const mapper = {
  bold: { format: "BOLD", type: "styles" },
  emphasis: { format: "ITALIC", type: "styles" },
  list: { format: "unordered-list-item", type: "blocks" },
  heading: { format: "header-two", type: "blocks" },
};

// Custom filter function to replace multiple spaces with a single space
function replaceMultipleSpaces(contentState) {
  const text = contentState.getText();
  const transformedText = text.replace(/\s+/g, " "); // Replace multiple spaces with a single space
  return contentState.merge({
    text: transformedText,
  });
}

export default function createFormatStripper(controls) {
  const filterConfiguration = {
    blocks: [],
    styles: [],
    entities: [pipedFilterConfig, linkFilterConfig],
    maxNesting: 0,
    whitespacedCharacters: [replaceMultipleSpaces],
  };

  mapKeys(controls, (value, key) => {
    if (mapper[key] && controls[key]) {
      filterConfiguration[mapper[key].type].push(mapper[key].format);
    }
  });

  return (editorState) => filterEditorState(filterConfiguration, editorState);
}
