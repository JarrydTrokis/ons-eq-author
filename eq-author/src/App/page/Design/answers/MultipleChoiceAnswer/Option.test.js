import React from "react";

import WrappingInput from "components/Forms/WrappingInput";
import { shallow, mount } from "enzyme";
import { StatelessOption } from "./Option";
import DeleteButton from "components/buttons/DeleteButton";
import { CHECKBOX, RADIO } from "constants/answer-types";

import { merge } from "lodash";
import createMockStore from "tests/utils/createMockStore";
import { useMutation } from "@apollo/react-hooks";
import { props } from "lodash/fp";

import { render as rtlRender, fireEvent, flushPromises, screen } from "tests/utils/rtl";


jest.mock("@apollo/react-hooks", () => ({
  useMutation: jest.fn(),
}));

useMutation.mockImplementation(jest.fn(() => [jest.fn()]));

// jest.mock("@apollo/react-hooks", () => ({
//   useMutation: () => [
//     jest.fn(
//       () =>
//         new Promise((resolve) =>
//           resolve({
//             data: {
//               createFolder: { id: "new-folder", pages: [{ id: "page-1" }] },
//             },
//           })
//         )
//     ),
//   ],
// }));

describe("Option", () => {
  let mockMutations;
  let mockEvent;
  let wrapper;
  let store;

  const option = {
    id: "1",
    label: "",
    description: "",
    additionalAnswer: {
      id: "additional1",
      label: "",
      type: "TextField",
    },
    __typename: "Option",
  };

  // const createWrapper = (otherProps, render = shallow) => {
  //   return render(<StatelessOption
  //           {...mockMutations}
  //           option={option}
  //           hasDeleteButton
  //           type={RADIO}
  //           store={store}
  //           {...otherProps}
  //         />);
  // };

  const render = (method = shallow, otherProps) => {
    wrapper = method(
      <StatelessOption
        {...mockMutations}
        option={option}
        hasDeleteButton
        type={RADIO}
        store={store}
        {...otherProps}
      />
    );

    return wrapper;
  };

  beforeEach(() => {

    // props = {
    //   onBlur: jest.fn(),
    //   onChange: jest.fn(),
    //   onUpdate: jest.fn(),
    //   onFocus: jest.fn(),
    //   onDelete: jest.fn(),
    //   onEnterKey: jest.fn(),
    // };

    mockEvent = {
      stopPropagation: jest.fn(),
      preventDefault: jest.fn(),
    };

    store = createMockStore();

    mockMutations = {
      onBlur: jest.fn(),
      onChange: jest.fn(),
      onUpdate: jest.fn(),
      onFocus: jest.fn(),
      onDelete: jest.fn(),
      onEnterKey: jest.fn(),
    };

    render();
  });

  it("should match snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("should render a checkbox", () => {
    render(mount, { type: CHECKBOX });
    expect(wrapper).toMatchSnapshot();
  });

  it("shouldn't render delete button if not applicable", () => {
    wrapper.setProps({ hasDeleteButton: false });
    expect(wrapper).toMatchSnapshot();
  });



  // it.only("should call onChange and onBlur correctly", async () => {
  //   const { getByTestId } = render(( otherProps) => <StatelessOption {...mockMutations}
  //     option={option}
  //     hasDeleteButton
  //     type={RADIO}
  //     store={store}
  //     {...otherProps}
  //    {...props} />);

  //   fireEvent.change(getByTestId("option-label"), {
  //     target: { value: "2" },
  //   });
  //   fireEvent.blur(getByTestId("number-input"));
  //   await flushPromises();
  //   expect(props.onBlur).toHaveBeenCalledWith(2);
  // });




  it.only("should call onChange on input", async() => {
    render(rtlRender, { type: CHECKBOX });
    // screen.debug();

    fireEvent.change(screen.getByTestId("option-label"), {
          target: { value: "2" },
        });
        // fireEvent.blur(screen.getByTestId("option-label"));
        // await flushPromises();
        expect(mockMutations.onChange).toHaveBeenCalledTimes(2);

    // const input = wrapper
    //   .find(`[data-test="option-label"]`)
    //   .first()
    //   .simulate("change");

    // wrapper
    // .find("[data-test='option-label']")
    // .first().simulate("change");
    // wrapper.find("[data-test='option-description']").first().simulate("change");

    // expect(mockMutations.onChange).toHaveBeenCalledTimes(1);
  });

  it("should update label on blur", () => {
    wrapper.find("[data-test='option-label']").simulate("blur", mockEvent);

    expect(mockMutations.onUpdate).toHaveBeenCalled();
  });

  it("should update description on blur", () => {
    wrapper
      .find("[data-test='option-description']")
      .simulate("blur", mockEvent);

    expect(mockMutations.onUpdate).toHaveBeenCalled();
  });

  it("should invoke onDelete callback when option deleted", () => {
    wrapper.find(DeleteButton).simulate("click", mockEvent);

    expect(mockMutations.onDelete).toHaveBeenCalledWith(option.id);
  });

  it("should call onEnterKey when Enter key pressed", () => {
    console.log(wrapper.debug({verbose:true}));

    wrapper
      .find(WrappingInput)
      .first()
      .simulate("keyDown", merge(mockEvent, { keyCode: 13 }));

    expect(mockMutations.onEnterKey).toHaveBeenCalledWith(mockEvent);
  });

  it("should not call onEnterKey when other keys are pressed", () => {
    wrapper
      .find(WrappingInput)
      .first()
      .simulate("keyDown", merge(mockEvent, { keyCode: 27 }));

    expect(mockMutations.onEnterKey).not.toHaveBeenCalled();
  });

  it("can turn off auto-focus", () => {
    wrapper = render(mount, { autoFocus: false });
    const input = wrapper
      .find(`[data-test="option-label"]`)
      .first()
      .getDOMNode();

    expect(input.hasAttribute("data-autofocus")).toBe(false);
  });
});
