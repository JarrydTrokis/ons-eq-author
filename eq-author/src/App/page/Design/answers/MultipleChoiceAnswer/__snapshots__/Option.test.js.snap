// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Option should match snapshot 1`] = `
<Option__StyledOption
  duration={200}
  hasDeleteButton={true}
  id="Option1"
  key="1"
  onDelete={[MockFunction]}
  onEnterKey={[MockFunction]}
  onFocus={[MockFunction]}
  store={
    Object {
      "dispatch": [MockFunction],
      "getState": [MockFunction],
      "subscribe": [MockFunction],
    }
  }
>
  <div>
    <Option__Flex>
      <MultipleChoice
        type="Radio"
      />
      <Option__OptionField>
        <Label
          bold={true}
          htmlFor="option-label-1"
        >
          Label
        </Label>
        <withChangeHandler(WrappingInput)
          bold={true}
          data-autofocus={true}
          data-test="option-label"
          id="option-label-1"
          name="label"
          onBlur={[MockFunction]}
          onChange={[MockFunction]}
          onKeyDown={[Function]}
          placeholder="Label"
          value=""
        />
      </Option__OptionField>
    </Option__Flex>
    <Option__OptionField>
      <Label
        bold={true}
        htmlFor="option-description-1"
      >
        Description (optional)
      </Label>
      <withChangeHandler(WrappingInput)
        data-test="option-description"
        id="option-description-1"
        name="description"
        onBlur={[MockFunction]}
        onChange={[MockFunction]}
        onKeyDown={[Function]}
        value=""
      />
    </Option__OptionField>
    <Option__ButtonsContainer>
      <Tooltip
        content="Move answer up"
        offset={
          Object {
            "bottom": 10,
            "top": 0,
          }
        }
        place="top"
      >
        <MoveButton
          data-test="btn-move-answer-up"
          disabled={true}
        >
          <Component />
        </MoveButton>
      </Tooltip>
      <Tooltip
        content="Move answer down"
        offset={
          Object {
            "bottom": 10,
            "top": 0,
          }
        }
        place="top"
      >
        <MoveButton
          data-test="btn-move-answer-down"
          disabled={true}
        >
          <Component />
        </MoveButton>
      </Tooltip>
      <Tooltip
        content="Delete option"
        offset={
          Object {
            "bottom": 10,
          }
        }
        place="top"
      >
        <DeleteButton
          aria-label="Delete option"
          data-test="btn-delete-option"
          disabled={false}
          onClick={[Function]}
          size="medium"
          type="button"
        />
      </Tooltip>
    </Option__ButtonsContainer>
  </div>
</Option__StyledOption>
`;

exports[`Option should render a checkbox 1`] = `
.c6 {
  font-weight: bold;
}

.c8 {
  font-weight: regular;
}

.c7 {
  font-size: 1em;
  border: 1px solid #999999;
  padding: 0.5em;
  color: #333333;
  display: block;
  width: 100%;
  -webkit-transition: outline-color 100ms ease-in,border-color 100ms ease-in;
  transition: outline-color 100ms ease-in,border-color 100ms ease-in;
  outline: 1px solid transparent;
  font-weight: inherit;
  resize: none;
  overflow: hidden;
}

.c7:hover {
  border-color: #3B7A9E;
  outline-color: #3B7A9E;
}

.c7:focus,
.c7:focus-within {
  border-color: #3B7A9E;
  outline-color: #3B7A9E;
  box-shadow: 0 0 0 3px #FDBD56;
}

.c7::-webkit-input-placeholder {
  color: #a3a3a3;
}

.c7::-moz-placeholder {
  color: #a3a3a3;
}

.c7:-ms-input-placeholder {
  color: #a3a3a3;
}

.c7::placeholder {
  color: #a3a3a3;
}

.c7:focus {
  outline: none;
  border: 1px solid #3B7A9E;
}

.c7[disabled] {
  opacity: 0.8;
  pointer-events: none;
}

.c5 {
  display: block;
  margin-bottom: 0.4em;
  font-weight: bold;
  vertical-align: middle;
  color: #333333;
  line-height: 1.3;
}

.c4 {
  display: block;
  width: 100%;
  margin-bottom: 1em;
  position: relative;
}

.c12 {
  color: #3B7A9E;
  border: none;
  background: transparent;
  cursor: pointer;
  -webkit-transition: color 200ms ease-in,opacity 300ms ease-in;
  transition: color 200ms ease-in,opacity 300ms ease-in;
  width: 1em;
  height: 1em;
  padding: 0;
  font-size: 2em;
}

.c12:hover {
  color: #333333;
}

.c12:focus {
  outline: 3px solid #FDBD56;
}

.c12[disabled] {
  opacity: 0.3;
  pointer-events: none;
}

.c13 {
  pointer-events: none;
  display: block;
  width: 1em;
  height: 1em;
}

.c13 path {
  fill: currentColor;
}

.c11 {
  background-color: black !important;
  opacity: 1 !important;
  font-size: 0.8rem !important;
  line-height: 1 !important;
  padding: 0.4rem 0.6rem !important;
  border-radius: 4px !important;
  white-space: pre;
}

.c10 {
  display: block;
  color: #3B7A9E;
  border: none;
  padding: 0;
  width: 2em;
  height: 2em;
  font-size: 1em;
  cursor: pointer;
  -webkit-transition: color 200ms ease-in,opacity 300ms ease-in;
  transition: color 200ms ease-in,opacity 300ms ease-in;
  background: transparent;
}

.c10 svg {
  pointer-events: none;
}

.c10 path {
  fill: currentColor;
}

.c10:hover {
  color: #333333;
}

.c10:focus {
  outline: 3px solid #FDBD56;
}

.c10[aria-disabled="true"] {
  pointer-events: none;
}

.c10[aria-disabled="true"] svg {
  opacity: 0.3;
}

.c2 {
  background: #E4E8EB;
  background-size: cover;
  height: 1.4em;
  width: 1.4em;
  display: inline-block;
  margin: 2em 1em 0 0;
  border-radius: 4px;
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
}

.c9 {
  position: absolute;
  right: 0;
  top: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  z-index: 2;
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
}

.c3 {
  margin-bottom: 1em;
}

.c0 {
  border: 1px solid #d6d8da;
  padding: 1em 1em 0;
  border-radius: 4px;
  position: relative;
  margin-bottom: 1em;
}

<StatelessOption
  autoFocus={true}
  getValidationError={[Function]}
  hasDeleteButton={true}
  labelPlaceholder="Label"
  onChange={[MockFunction]}
  onDelete={[MockFunction]}
  onEnterKey={[MockFunction]}
  onFocus={[MockFunction]}
  onUpdate={[MockFunction]}
  option={
    Object {
      "__typename": "Option",
      "description": "",
      "id": "1",
      "label": "",
    }
  }
  store={
    Object {
      "dispatch": [MockFunction],
      "getState": [MockFunction],
      "subscribe": [MockFunction],
    }
  }
  type="Checkbox"
>
  <Option__StyledOption
    duration={200}
    hasDeleteButton={true}
    id="Option1"
    key="1"
    onDelete={[MockFunction]}
    onEnterKey={[MockFunction]}
    onFocus={[MockFunction]}
    store={
      Object {
        "dispatch": [MockFunction],
        "getState": [MockFunction],
        "subscribe": [MockFunction],
      }
    }
  >
    <div
      className="c0"
      id="Option1"
      onFocus={[MockFunction]}
    >
      <div>
        <Option__Flex>
          <div
            className="c1"
          >
            <MultipleChoice
              type="Checkbox"
            >
              <div
                className="c2"
                type="Checkbox"
              />
            </MultipleChoice>
            <Option__OptionField>
              <Field
                className="c3"
                disabled={false}
                last={false}
              >
                <Field__StyledField
                  className="c3"
                  disabled={false}
                  last={false}
                >
                  <div
                    className="c3 c4"
                    disabled={false}
                  >
                    <Label
                      bold={true}
                      htmlFor="option-label-1"
                    >
                      <Label__StyledLabel
                        bold={true}
                        htmlFor="option-label-1"
                      >
                        <label
                          className="c5"
                          htmlFor="option-label-1"
                        >
                          Label
                        </label>
                      </Label__StyledLabel>
                    </Label>
                    <withChangeHandler(WrappingInput)
                      bold={true}
                      data-autofocus={true}
                      data-test="option-label"
                      id="option-label-1"
                      name="label"
                      onBlur={[MockFunction]}
                      onChange={[MockFunction]}
                      onKeyDown={[Function]}
                      placeholder="Label"
                      value=""
                    >
                      <WrappingInput
                        bold={true}
                        data-autofocus={true}
                        data-test="option-label"
                        id="option-label-1"
                        name="label"
                        onBlur={[MockFunction]}
                        onChange={[Function]}
                        onKeyDown={[Function]}
                        placeholder="Label"
                        value=""
                      >
                        <WrappingInput__StyleContext
                          bold={true}
                          onChange={[Function]}
                          onKeyDown={[Function]}
                        >
                          <div
                            className="c6"
                            onChange={[Function]}
                            onKeyDown={[Function]}
                          >
                            <WrappingInput__TextArea
                              aria-invalid="false"
                              data-autofocus={true}
                              data-test="option-label"
                              id="option-label-1"
                              name="label"
                              onBlur={[MockFunction]}
                              onChange={[Function]}
                              onKeyDown={[Function]}
                              placeholder="Label"
                              value=""
                            >
                              <TextareaAutosize
                                aria-invalid="false"
                                className="c7"
                                data-autofocus={true}
                                data-test="option-label"
                                id="option-label-1"
                                inputRef={[Function]}
                                name="label"
                                onBlur={[MockFunction]}
                                onChange={[Function]}
                                onHeightChange={[Function]}
                                onKeyDown={[Function]}
                                placeholder="Label"
                                useCacheForDOMMeasurements={false}
                                value=""
                              >
                                <textarea
                                  aria-invalid="false"
                                  className="c7"
                                  data-autofocus={true}
                                  data-test="option-label"
                                  id="option-label-1"
                                  name="label"
                                  onBlur={[MockFunction]}
                                  onChange={[Function]}
                                  onKeyDown={[Function]}
                                  placeholder="Label"
                                  style={
                                    Object {
                                      "height": 0,
                                    }
                                  }
                                  value=""
                                />
                              </TextareaAutosize>
                            </WrappingInput__TextArea>
                          </div>
                        </WrappingInput__StyleContext>
                      </WrappingInput>
                    </withChangeHandler(WrappingInput)>
                  </div>
                </Field__StyledField>
              </Field>
            </Option__OptionField>
          </div>
        </Option__Flex>
        <Option__OptionField>
          <Field
            className="c3"
            disabled={false}
            last={false}
          >
            <Field__StyledField
              className="c3"
              disabled={false}
              last={false}
            >
              <div
                className="c3 c4"
                disabled={false}
              >
                <Label
                  bold={true}
                  htmlFor="option-description-1"
                >
                  <Label__StyledLabel
                    bold={true}
                    htmlFor="option-description-1"
                  >
                    <label
                      className="c5"
                      htmlFor="option-description-1"
                    >
                      Description (optional)
                    </label>
                  </Label__StyledLabel>
                </Label>
                <withChangeHandler(WrappingInput)
                  data-test="option-description"
                  id="option-description-1"
                  name="description"
                  onBlur={[MockFunction]}
                  onChange={[MockFunction]}
                  onKeyDown={[Function]}
                  value=""
                >
                  <WrappingInput
                    bold={false}
                    data-test="option-description"
                    id="option-description-1"
                    name="description"
                    onBlur={[MockFunction]}
                    onChange={[Function]}
                    onKeyDown={[Function]}
                    value=""
                  >
                    <WrappingInput__StyleContext
                      bold={false}
                      onChange={[Function]}
                      onKeyDown={[Function]}
                    >
                      <div
                        className="c8"
                        onChange={[Function]}
                        onKeyDown={[Function]}
                      >
                        <WrappingInput__TextArea
                          aria-invalid="false"
                          data-test="option-description"
                          id="option-description-1"
                          name="description"
                          onBlur={[MockFunction]}
                          onChange={[Function]}
                          onKeyDown={[Function]}
                          value=""
                        >
                          <TextareaAutosize
                            aria-invalid="false"
                            className="c7"
                            data-test="option-description"
                            id="option-description-1"
                            inputRef={[Function]}
                            name="description"
                            onBlur={[MockFunction]}
                            onChange={[Function]}
                            onHeightChange={[Function]}
                            onKeyDown={[Function]}
                            useCacheForDOMMeasurements={false}
                            value=""
                          >
                            <textarea
                              aria-invalid="false"
                              className="c7"
                              data-test="option-description"
                              id="option-description-1"
                              name="description"
                              onBlur={[MockFunction]}
                              onChange={[Function]}
                              onKeyDown={[Function]}
                              style={
                                Object {
                                  "height": 0,
                                }
                              }
                              value=""
                            />
                          </TextareaAutosize>
                        </WrappingInput__TextArea>
                      </div>
                    </WrappingInput__StyleContext>
                  </WrappingInput>
                </withChangeHandler(WrappingInput)>
              </div>
            </Field__StyledField>
          </Field>
        </Option__OptionField>
        <Option__ButtonsContainer>
          <div
            className="c9"
          >
            <Tooltip
              content="Move answer up"
              offset={
                Object {
                  "bottom": 10,
                  "top": 0,
                }
              }
              place="top"
            >
              <MoveButton
                data-for="tooltip-1"
                data-test="btn-move-answer-up"
                data-tip={true}
                disabled={true}
              >
                <MoveButton__Button
                  aria-disabled={true}
                  data-for="tooltip-1"
                  data-test="btn-move-answer-up"
                  data-tip={true}
                >
                  <button
                    aria-disabled={true}
                    className="c10"
                    data-for="tooltip-1"
                    data-test="btn-move-answer-up"
                    data-tip={true}
                  >
                    <Component>
                      <svg />
                    </Component>
                  </button>
                </MoveButton__Button>
              </MoveButton>
              <Tooltip__StyledTooltip
                delayShow={200}
                effect="solid"
                id="tooltip-1"
                innerRef={[Function]}
                offset={
                  Object {
                    "bottom": 10,
                    "top": 0,
                  }
                }
                place="top"
              >
                <ReactTooltip
                  className="c11"
                  clickable={false}
                  delayShow={200}
                  effect="solid"
                  id="tooltip-1"
                  insecure={true}
                  offset={
                    Object {
                      "bottom": 10,
                      "top": 0,
                    }
                  }
                  place="top"
                  resizeHide={true}
                  wrapper="div"
                >
                  <div
                    className="__react_component_tooltip place-top type-dark "
                    data-id="tooltip"
                    id="tooltip-1"
                  >
                    Move answer up
                  </div>
                </ReactTooltip>
              </Tooltip__StyledTooltip>
            </Tooltip>
            <Tooltip
              content="Move answer down"
              offset={
                Object {
                  "bottom": 10,
                  "top": 0,
                }
              }
              place="top"
            >
              <MoveButton
                data-for="tooltip-2"
                data-test="btn-move-answer-down"
                data-tip={true}
                disabled={true}
              >
                <MoveButton__Button
                  aria-disabled={true}
                  data-for="tooltip-2"
                  data-test="btn-move-answer-down"
                  data-tip={true}
                >
                  <button
                    aria-disabled={true}
                    className="c10"
                    data-for="tooltip-2"
                    data-test="btn-move-answer-down"
                    data-tip={true}
                  >
                    <Component>
                      <svg />
                    </Component>
                  </button>
                </MoveButton__Button>
              </MoveButton>
              <Tooltip__StyledTooltip
                delayShow={200}
                effect="solid"
                id="tooltip-2"
                innerRef={[Function]}
                offset={
                  Object {
                    "bottom": 10,
                    "top": 0,
                  }
                }
                place="top"
              >
                <ReactTooltip
                  className="c11"
                  clickable={false}
                  delayShow={200}
                  effect="solid"
                  id="tooltip-2"
                  insecure={true}
                  offset={
                    Object {
                      "bottom": 10,
                      "top": 0,
                    }
                  }
                  place="top"
                  resizeHide={true}
                  wrapper="div"
                >
                  <div
                    className="__react_component_tooltip place-top type-dark "
                    data-id="tooltip"
                    id="tooltip-2"
                  >
                    Move answer down
                  </div>
                </ReactTooltip>
              </Tooltip__StyledTooltip>
            </Tooltip>
            <Tooltip
              content="Delete option"
              offset={
                Object {
                  "bottom": 10,
                }
              }
              place="top"
            >
              <DeleteButton
                aria-label="Delete option"
                data-for="tooltip-3"
                data-test="btn-delete-option"
                data-tip={true}
                disabled={false}
                onClick={[Function]}
                size="medium"
                type="button"
              >
                <DeleteButton__StyledDeleteButton
                  aria-label="Delete option"
                  data-for="tooltip-3"
                  data-test="btn-delete-option"
                  data-tip={true}
                  disabled={false}
                  onClick={[Function]}
                  size="medium"
                  type="button"
                >
                  <button
                    aria-label="Delete option"
                    className="c12"
                    data-for="tooltip-3"
                    data-test="btn-delete-option"
                    data-tip={true}
                    disabled={false}
                    onClick={[Function]}
                    size="medium"
                    type="button"
                  >
                    <DeleteButton__CloseIcon>
                      <Component
                        className="c13"
                      >
                        <svg />
                      </Component>
                    </DeleteButton__CloseIcon>
                  </button>
                </DeleteButton__StyledDeleteButton>
              </DeleteButton>
              <Tooltip__StyledTooltip
                delayShow={200}
                effect="solid"
                id="tooltip-3"
                innerRef={[Function]}
                offset={
                  Object {
                    "bottom": 10,
                  }
                }
                place="top"
              >
                <ReactTooltip
                  className="c11"
                  clickable={false}
                  delayShow={200}
                  effect="solid"
                  id="tooltip-3"
                  insecure={true}
                  offset={
                    Object {
                      "bottom": 10,
                    }
                  }
                  place="top"
                  resizeHide={true}
                  wrapper="div"
                >
                  <div
                    className="__react_component_tooltip place-top type-dark "
                    data-id="tooltip"
                    id="tooltip-3"
                  >
                    Delete option
                  </div>
                </ReactTooltip>
              </Tooltip__StyledTooltip>
            </Tooltip>
          </div>
        </Option__ButtonsContainer>
      </div>
    </div>
  </Option__StyledOption>
</StatelessOption>
`;

exports[`Option shouldn't render delete button if not applicable 1`] = `
<Option__StyledOption
  duration={200}
  hasDeleteButton={false}
  id="Option1"
  key="1"
  onDelete={[MockFunction]}
  onEnterKey={[MockFunction]}
  onFocus={[MockFunction]}
  store={
    Object {
      "dispatch": [MockFunction],
      "getState": [MockFunction],
      "subscribe": [MockFunction],
    }
  }
>
  <div>
    <Option__Flex>
      <MultipleChoice
        type="Radio"
      />
      <Option__OptionField>
        <Label
          bold={true}
          htmlFor="option-label-1"
        >
          Label
        </Label>
        <withChangeHandler(WrappingInput)
          bold={true}
          data-autofocus={true}
          data-test="option-label"
          id="option-label-1"
          name="label"
          onBlur={[MockFunction]}
          onChange={[MockFunction]}
          onKeyDown={[Function]}
          placeholder="Label"
          value=""
        />
      </Option__OptionField>
    </Option__Flex>
    <Option__OptionField>
      <Label
        bold={true}
        htmlFor="option-description-1"
      >
        Description (optional)
      </Label>
      <withChangeHandler(WrappingInput)
        data-test="option-description"
        id="option-description-1"
        name="description"
        onBlur={[MockFunction]}
        onChange={[MockFunction]}
        onKeyDown={[Function]}
        value=""
      />
    </Option__OptionField>
    <Option__ButtonsContainer>
      <Tooltip
        content="Move answer up"
        offset={
          Object {
            "bottom": 10,
            "top": 0,
          }
        }
        place="top"
      >
        <MoveButton
          data-test="btn-move-answer-up"
          disabled={true}
        >
          <Component />
        </MoveButton>
      </Tooltip>
      <Tooltip
        content="Move answer down"
        offset={
          Object {
            "bottom": 10,
            "top": 0,
          }
        }
        place="top"
      >
        <MoveButton
          data-test="btn-move-answer-down"
          disabled={true}
        >
          <Component />
        </MoveButton>
      </Tooltip>
      <Tooltip
        content="Delete option"
        offset={
          Object {
            "bottom": 10,
          }
        }
        place="top"
      >
        <DeleteButton
          aria-label="Delete option"
          data-test="btn-delete-option"
          disabled={true}
          onClick={[Function]}
          size="medium"
          type="button"
        />
      </Tooltip>
    </Option__ButtonsContainer>
  </div>
</Option__StyledOption>
`;
